### PR TITLE
feat-documentation

### PR DESCRIPTION
**Pull request by KevinMB0220**
### 📝 Summary 
- Closes #(issue)
- Completes all documentation for the Battle feature.
### 🔄 Changes Made
- Updated documentation in battle.cairo under systems, modules, and test directories.
- Added comprehensive comments to methods and classes within each file.
- Included necessary explanations to improve code readability.
### 🔧 Tests Results
- All unit tests passed successfully, covering the documented Battle functionality.

### 🔜 Next Steps
- Review if any additional documentation adjustments are needed.
- Confirm no further validation tests are required.

PoC inicial

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Battle` struct to manage battle states between players.
	- Enhanced the README with installation instructions and a guide for setting up Git hooks.
	
- **Bug Fixes**
	- Improved test coverage for battle interactions and initialization.
	
- **Documentation**
	- Added extensive comments throughout the codebase to clarify functionality and improve readability.
	- Updated contribution guidelines for clarity and accessibility.

- **Chores**
	- Reformatted various configuration files for consistency.

### CODE REVIEW

File: src/systems/battle.cairo [hunk 1]
```diff
@@ -61,117 +97,153 @@ mod battle_system {
...
             player_beast = get!(world, battle.active_beast_player, (Beast));
 
+            // Emit an event to notify that the potion has been used
             let message = 'Item Used!';
             emit!(world, (StatusBattle { battle_id,  message }));
         }
 
+// ***************************************************************
+//                           Flee Action
+// ***************************************************************
```
Comment (danielcdz) [inline src/systems/battle.cairo:227]: Can you delete these in all the functions?

**Review by danielcdz — CHANGES_REQUESTED** [review]

**Comment by danielcdz** [comment]
You are missing the documentation for the system tests, these are inside tests directory

**Review by danielcdz — APPROVED** [review]
LGTM

**Review by coxmars — APPROVED** [review]
LGTM

**Comment by danielcdz** [comment]
Why am I seeing a lot of files that are not required to receive any kind of update modified?

**Pull request closed without merging**